File: apps/web/src/lib/format.ts
// Display helpers — Pakistan locale defaults.

export function formatPKR(amount: number): string {
  return new Intl.NumberFormat("en-PK", {
    style: "currency",
    currency: "PKR",
    maximumFractionDigits: 0,
  }).format(amount);
}

export function formatTime(iso?: string): string {
  if (!iso) return "—";
  return new Intl.DateTimeFormat("en-PK", {
    hour: "numeric",
    minute: "2-digit",
    hour12: true,
  }).format(new Date(iso));
}

export function formatDate(iso?: string): string {
  if (!iso) return "—";
  return new Intl.DateTimeFormat("en-PK", {
    weekday: "short",
    day: "numeric",
    month: "short",
  }).format(new Date(iso));
}

export function formatDuration(min?: number): string {
  if (!min) return "—";
  const h = Math.floor(min / 60);
  const m = min % 60;
  return [h ? `${h}h` : "", m ? `${m}m` : ""].filter(Boolean).join(" ");
}

File: apps/mobile/lib/format.dart
import 'package:intl/intl.dart';

final _pkr = NumberFormat.currency(locale: 'en_PK', symbol: 'Rs ', decimalDigits: 0);

String formatPKR(num amount) => _pkr.format(amount);

String formatTime(String? iso) {
  if (iso == null) return '—';
  final dt = DateTime.parse(iso).toLocal();
  return DateFormat('h:mm a').format(dt);
}

String formatDuration(int? minutes) {
  if (minutes == null) return '—';
  final h = minutes ~/ 60;
  final m = minutes % 60;
  return [if (h > 0) '${h}h', if (m > 0) '${m}m'].join(' ');
}
